lib/curl_setup.h

```diff
@@ -838,6 +838,10 @@ int getpwuid_r(uid_t uid, struct passwd *pwd, char *buf,
 #define USE_HTTP3
 #endif
 
+#if defined(HAVE_BASENAME) && defined(WIN32)
```

Comment:
I suggest a brief comment explaining why. Also is this a bug in mingw-w64 or is it our bug?

Reply:
Added a comment.

Sticking with the short example from https://github.com/curl/curl/issues/10261#issuecomment-1426679328, this seems to be a mingw-w64 issue. But hard to say for certain without reproducing/debugging this with that exact (raw byte!) input and locale (and possibly the OS env). Also the original bug report is a slightly different case.

[ Just by eyeballing the source, there seems to be one bug at least, where the `if (*refname)` condition is always true, so the `else` branch never executes, even though it should if the input has no pathsep. The `else` branch should normally be hit by the short example, but not by the original bug report. ]